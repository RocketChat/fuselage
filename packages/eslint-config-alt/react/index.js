module.exports = {
  extends: '../minimal',
  plugins: ['react', 'react-hooks'],
  parser: '@babel/eslint-parser',
  rules: {
    'jsx-quotes': ['error', 'prefer-single'],
    'react/display-name': 'error',
    'react/jsx-uses-react': 'error',
    'react/jsx-uses-vars': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-fragments': ['error', 'syntax'],
    'react/no-multi-comp': 'error',
    'react/react-in-jsx-scope': 'error',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
    'react': {
      version: 'detect',
    },
  },
  env: {
    browser: true,
    es6: true,
  },
  overrides: [
    {
      files: ['**/*.ts', '**/*.tsx'],
      extends: '../typescript',
      parser: '@typescript-eslint/parser',
      plugins: ['react', 'react-hooks'],
      rules: {
        'jsx-quotes': ['error', 'prefer-single'],
        'react/display-name': 'error',
        'react/jsx-uses-react': 'error',
        'react/jsx-uses-vars': 'error',
        'react/jsx-no-undef': 'error',
        'react/jsx-fragments': ['error', 'syntax'],
        'react/no-multi-comp': 'error',
        'react/react-in-jsx-scope': 'error',
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
      },
      env: {
        browser: true,
        es6: true,
      },
      settings: {
        'import/resolver': {
          node: {
            extensions: ['.js', '.jsx', '.ts', '.tsx'],
          },
        },
        'react': {
          version: 'detect',
        },
      },
    },
    {
      files: [
        '**/*.stories.js',
        '**/*.stories.jsx',
        '**/*.stories.ts',
        '**/*.stories.tsx',
      ],
      rules: {
        'react/display-name': 'off',
        'react/no-multi-comp': 'off',
      },
    },
  ],
};
