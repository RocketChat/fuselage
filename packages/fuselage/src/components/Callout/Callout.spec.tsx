import { render } from '@testing-library/react';
import React from 'react';

import { Callout } from '.';

it('renders without crashing', () => {
  render(<Callout>Callout</Callout>);
});
