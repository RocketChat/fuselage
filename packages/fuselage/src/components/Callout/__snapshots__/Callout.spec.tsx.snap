// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CheckBox Rendering] renders CustomIcon without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout undefined "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-hash rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__title"
        >
          This is a message with custom icon
        </div>
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Danger without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout rcx-callout--danger "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-ban rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__title"
        >
          This is a danger message
        </div>
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout undefined "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__title"
        >
          This is a generic title
        </div>
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Info without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout rcx-callout--info "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__title"
        >
          This is a info message
        </div>
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Success without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout rcx-callout--success "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-checkmark-circled rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__title"
        >
          This is a success message
        </div>
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders Warning without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout rcx-callout--warning "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-warning rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__title"
        >
          This is a warning message
        </div>
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;

exports[`[CheckBox Rendering] renders WithDescriptionOnly without crashing 1`] = `
<body>
  <div>
    <section
      class="rcx-box rcx-box--full  rcx-callout undefined "
    >
      <i
        aria-hidden="true"
        class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon  rcx-callout__icon rcx-css-4pvxx3"
      >
        
      </i>
      <div
        class="rcx-box rcx-box--full rcx-callout__wrapper"
      >
        <div
          class="rcx-box rcx-box--full rcx-callout__content"
        >
          This is a generic description.
        </div>
      </div>
    </section>
  </div>
</body>
`;
